# HG changeset patch
# User Rik <rik@octave.org>
# Date 1355768168 28800
#      Mon Dec 17 10:16:08 2012 -0800
# Node ID 420b5603cbbfb5c11f1e9d2c9f5a7a1f26c2f4be
# Parent  b15cfb7ed0eb4493ce618d25353220f01f13dee8
build: Move usejava.m from scripts/miscellaneous to scripts/java directory.

* scripts/java/module.mk: Add usejava.m to build system.

* scripts/java/usejava.m: Added m-file.

* scripts/miscellaneous/module.mk: Remove usejava from miscellaneous directory build.

* scripts/miscellaneous/usejava.m: Removed m-file.

diff --git a/scripts/java/module.mk b/scripts/java/module.mk
--- a/scripts/java/module.mk
+++ b/scripts/java/module.mk
@@ -8,16 +8,17 @@ java_FCN_FILES = \
   java/javaArray.m \
   java/javaaddpath.m \
   java/javaclasspath.m \
   java/javamem.m \
   java/javarmpath.m \
   java/listdlg.m \
   java/msgbox.m \
   java/questdlg.m \
+  java/usejava.m \
   java/warndlg.m
 
 FCN_FILES += $(java_FCN_FILES)
 
 PKG_ADD_FILES += java/PKG_ADD
 
 DIRSTAMP_FILES += java/$(octave_dirstamp)
 
diff --git a/scripts/miscellaneous/usejava.m b/scripts/java/usejava.m
rename from scripts/miscellaneous/usejava.m
rename to scripts/java/usejava.m
--- a/scripts/miscellaneous/usejava.m
+++ b/scripts/java/usejava.m
@@ -33,57 +33,60 @@
 ## @item "jvm"
 ## Java Virtual Machine.
 ##
 ## @item "swing"
 ## Swing components for lightweight GUIs.
 ## @end table
 ##
 ## @code{usejava} determines if specific Java features are available in an
-## Octave session.  This function is provided for compatibility with scripts
-## which may alter their behavior based on the availability of Java.  The
-## feature "desktop" always returns @code{false} as Octave has no Java-based
-## desktop.  Other features may be available if the Octave-Forge Java package
-## has been installed.
+## Octave session.  This function is provided for scripts which may alter
+## their behavior based on the availability of Java.  The feature "desktop"
+## always returns @code{false} as Octave has no Java-based desktop.  Other
+## features may be available if Octave was compiled with the Java Interface
+## and Java is installed.
 ## @end deftypefn
 
 function retval = usejava (feature)
 
   if (nargin != 1 || ! ischar (feature))
     print_usage ();
   endif
 
   retval = false;
 
   switch feature
-    ## For each feature, try javamethods() on a Java class of a feature
+    ## For each feature, try methods() on a Java class of a feature
     case "awt"
       try
-        dum = javamethods ("java.awt.Frame");
+        dum = methods ("java.awt.Frame");
         retval = true;
       end_try_catch
     case "desktop"
       ## Octave has no Java based GUI/desktop, leave retval = false
     case "jvm"
       try
-        dum = javamethods ("java.lang.Runtime");
+        dum = methods ("java.lang.Runtime");
         retval = true;
       end_try_catch
     case "swing"
       try
-        dum = javamethods ("javax.swing.Popup");
+        dum = methods ("javax.swing.Popup");
         retval = true;
       end_try_catch
     otherwise
       error ("usejava: unrecognized feature '%s'", feature);
   endswitch
 
 endfunction
 
 
 %!assert (usejava ("desktop"), false)
 
+%!testif HAVE_JAVA
+%! assert (usejava ("jvm"), true);
+
 %% Test input validation
 %!error usejava ()
 %!error usejava (1, 2)
 %!error usejava (1)
 %!error <unrecognized feature> usejava ("abc")
 
diff --git a/scripts/miscellaneous/module.mk b/scripts/miscellaneous/module.mk
--- a/scripts/miscellaneous/module.mk
+++ b/scripts/miscellaneous/module.mk
@@ -64,17 +64,16 @@ miscellaneous_FCN_FILES = \
   miscellaneous/symvar.m \
   miscellaneous/tar.m \
   miscellaneous/tempdir.m \
   miscellaneous/tempname.m \
   miscellaneous/unix.m \
   miscellaneous/unpack.m \
   miscellaneous/untar.m \
   miscellaneous/unzip.m \
-  miscellaneous/usejava.m \
   miscellaneous/ver.m \
   miscellaneous/version.m \
   miscellaneous/warning_ids.m \
   miscellaneous/what.m \
   miscellaneous/xor.m \
   miscellaneous/zip.m \
   $(miscellaneous_PRIVATE_FCN_FILES)
 
